### DISCUSSION TITLE
Integration with CTS600 connect via Waveshare rs485 to Wifi

### DISCUSSION THREAD

**Discussion by Lutehi-Gregor** [veista/nilan#119] [Q&A]
Dear developer
Our house has a ventilation system from Nilan.
The control system installed is a CTS600.
It also has an RS485 interface to the control panel.
In the meantime, I have purchased the converter from [Waveshare ](https://www.waveshare.com/wiki/RS485_TO_WIFI/ETH).
Unfortunately, I still can't get a connection to my Nilan control unit. See also the photos in the appendix.
I hope you can help me.
Thank you in advance for your support.
Kind regards
Gregor Lüthi
![IMG_8583](https://github.com/veista/nilan/assets/61920830/1365b8fa-24e5-4fd1-a4f6-d8a1b196848c)
![IMG_8581](https://github.com/veista/nilan/assets/61920830/329d630d-6f62-4480-8620-6e7718146377)
![IMG_8614](https://github.com/veista/nilan/assets/619
[Uploading IMG_8590.zip…]()
20830/101d304d-690a-4432-84d6-00c0dfb187f5)

**Comment by veista** [comment]
Cts600 is not supported nor can it be implemented to this integration.

**Reply by Lutehi-Gregor** [reply]
Thank you for the quick reply.
What a pity - could you possibly do something with this [integration](https://github.com/frodef/nilan-cts600-homeassistant)? Should be suitable for the CTS600 but does not have an IP interface.
Many thanks in advance for your expert reply.

**Reply by veista** [reply]
#49

**Reply by veista** [reply]
Yes, you can use that integration. The problem is that the button presses can't be emulated with a LAN connection. With a wired USB connection it should work.

**Reply by veista** [reply]
Also I don't think your waveshare adapter is the correct one for this integration or the one you linked.

**Reply by Lutehi-Gregor** [reply]
OK, USB works LAN or WLAN does not work.
Currently incomprehensible to me.
Why should the waveshare not work.
It emulates RS485 to the following protocols around TCP / UDP / ARP / ICMP / DHCP / DNS / HTTP / MQTT.
But I can see that my idea of making our ventilation system smart is going to be a bigger challenge than I thought.

**Reply by veista** [reply]
Well: This integration requires a Modbus RTU to Modbus TCP translator on LAN connection. It is a bridge between two different protocols. USB to RS485 adapter uses Modbus RTU so no translation needed.

The button translations can't be emulated on LAN, because of latency and delay inconsistency. Sometimes there is one press other times 100. This is an unfortunate design choice implemented by Nilan.

It is not hard to create hardware to replace the display using esp32, raspberry picow etc. And implementing Mqtt or modbus TCP with a display and physical buttons. It requires a lot of time and involvement.

**Reply by Lutehi-Gregor** [reply]
OK, thanks for the explanation regarding the buttons.
My idea would be to do the complete control via hassio.
In other words, also settings such as the weekly programmes which are made via the control unit.
What is the problem, the control board (CTS600) of the control unit or the software installed?
The Waveshare should be able to do the translation you mentioned.
See the photo below.
The problem is the distance. That is why I found the solution via WLAN practicable. 
Of course, I could put the hassio control centre in the cellar and test the integration via USB.
But I would have to test whether the Z-Wave and ZigBee signals reach that far.
An alternative would be to integrate a second hassio station - whether that is possible...
![Waveshare rs485 to Wifi](https://github.com/veista/nilan/assets/61920830/f15ea5c6-c186-4de9-b6f2-16e7c3f4c4cc)

**Reply by veista** [reply]
You would be better off just using a raspberry pi to replace the display and making an mqtt interface on that. You could also make it so you keep the original panel.

**Reply by veista** [reply]
The problem is that cts600 has only one rs485 connection and it is for the panel. It also lacks the more open software so you have to emulate button presses to do any sort of configuring.

**Reply by veista** [reply]
Also keep in mind that i do not own any of these older devices so this is just my understanding.

**Reply by Lutehi-Gregor** [reply]
Many thanks for the additions.
I should actually cancel the project.
Without the function of the buttons, remote diagnosis via hassio makes no sense.
Nevertheless, I have now ordered a USB to RS485 serial converter, as they don't cost the earth. Maybe it will work.
I don't understand the reference to Raspberry pi and the original panel. I can only control one device per RS485. In other words, either the Raspberry with the mqtt software or the display from Nilan?

**Reply by veista** [reply]
Yes, but you can create a device that forwards messages from the cts600 card to the panel and also sends messages from ha via mqtt to the cts600.

**Reply by Lutehi-Gregor** [reply]
Good morning
Thank you for the feedback on this bridge.
My programming skills are not very good. Are there already approaches to this?
Regarding hardware, there are [RS-485 splitters](https://ewimar.com/distributor-splitter-rs-485-bus.html). Could I use such a device to separate the signal and thus forward both the original operation and the signal to hassio?
Many thanks in advance for your support in the possible realisation of the integration of Nilan.

**Reply by veista** [reply]
I have no experience on such devices, possibly yes, but it does not make it much easier if you want remote buttons Via Lan. If you only need to change ventilation etc  remotely it would be great.

**Reply by Lutehi-Gregor** [reply]
OK, I have now put my hassio in the cellar. Everything is currently running - so I should possibly have success with the RS485 to USB cable.
If I still need the original control unit, I will try an RS485 converter.

**Reply by veista** [reply]
Ok. Good luck!
